From: tonyj@suse.de
Subject: Export namespace semaphore
Patch-mainline: no

Export global namespace_sem (this used to be a per namespace semaphore).
Alas, this isn't going to win _any_ points for style.
Patch is not in mainline -- pending AppArmor code submission to lkml


---
 fs/namespace.c                |    3 ++-
 include/linux/mnt_namespace.h |    3 +++
 2 files changed, 5 insertions(+), 1 deletion(-)

--- linux-2.6.19.orig/fs/namespace.c
+++ linux-2.6.19/fs/namespace.c
@@ -37,7 +37,8 @@ static int event;
 static struct list_head *mount_hashtable __read_mostly;
 static int hash_mask __read_mostly, hash_bits __read_mostly;
 static struct kmem_cache *mnt_cache __read_mostly;
-static struct rw_semaphore namespace_sem;
+struct rw_semaphore namespace_sem;
+EXPORT_SYMBOL_GPL(namespace_sem);
 
 /* /sys/fs */
 decl_subsys(fs, NULL, NULL);
--- linux-2.6.19.orig/include/linux/mnt_namespace.h
+++ linux-2.6.19/include/linux/mnt_namespace.h
@@ -6,6 +6,9 @@
 #include <linux/sched.h>
 #include <linux/nsproxy.h>
 
+/* exported for AppArmor (SubDomain) */
+extern struct rw_semaphore namespace_sem;
+
 struct mnt_namespace {
 	atomic_t		count;
 	struct vfsmount *	root;
